packages/PriceLockup/__tests__/PriceLockup.spec.jsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react'
+import { shallow, mount } from 'enzyme'
+import Text from '@tds/core-text'
+import HairlineDivider from '@tds/core-hairline-divider'
+import PriceLockup from '../PriceLockup'
+
+const defaultProps = {
+  signDirection: 'left',
+  price: '25',
+  rateText: '/month',
+  bottomText: 'bottom text',
+}
+
+describe('PriceLockup', () => {
+  const doShallow = (props = {}) => shallow(<PriceLockup {...defaultProps} {...props} />)
+  const doMount = (props = {}) => mount(<PriceLockup {...defaultProps} {...props} />)
+
+  it('renders with default props', () => {
+    const priceLockup = doShallow()
+    expect(priceLockup).toMatchSnapshot()
+  })
+
+  it('does not allow custom CSS', () => {
+    const priceLockup = doShallow({
+      className: 'my-custom-class',
+      style: { color: 'hotpink' },
+    })
+    expect(priceLockup).not.toHaveProp('className', 'my-custom-class')
+    expect(priceLockup).not.toHaveProp('style')
+  })
+
+  describe('size prop is small', () => {
+    const priceLockup = doShallow({ size: 'small' })
+    it('will make price h2 sized', () => {
+      expect(priceLockup.html()).toContain('priceValueSignsmall')
+    })
+    it('will make rate text medium sized', () => {
+      priceLockup.setProps({ rateText: '/month' })
+      expect(priceLockup).toContainReact(<Text size="medium">/month</Text>)
+    })
+    it('will make top text small sized', () => {
+      priceLockup.setProps({ topText: 'top text' })
+      expect(priceLockup).toContainReact(<Text size="small">top text</Text>)
+    })
+  })
+
+  describe('size prop is medium', () => {
+    const priceLockup = doShallow({ size: 'medium' })
+    it('will make price h2 sized', () => {
+      expect(priceLockup.html()).toContain('priceValueSignmedium')
+    })
+    it('will make rate text medium sized', () => {
+      priceLockup.setProps({ rateText: '/month' })
+      expect(priceLockup).toContainReact(<Text size="medium">/month</Text>)
+    })
+    it('will make top text small sized', () => {
+      priceLockup.setProps({ topText: 'top text' })
+      expect(priceLockup).toContainReact(<Text size="small">top text</Text>)
+    })
+    it('will make bottom text medium sized', () => {
+      priceLockup.setProps({ bottomText: 'bottom text' })
+      expect(priceLockup).toContainReact(<Text size="medium">bottom text</Text>)
+    })
+  })
+
+  describe('size prop is large', () => {
+    const priceLockup = doShallow({ size: 'large' })
+    it('will make DollarSign h1 sized', () => {
+      expect(priceLockup.html()).toContain('dollarSignH1Style')
```

Comment:
This test checks if the string `dollarSignH1Style` exists anywhere in the component, but not whether it was applied to the dollar sign. For a more reliable test, you can add `data-testid="dollarSign"` to the dollar sign element and find it in Jest:

```jsx
expect(priceLockup.find('[data-testid="dollarSign"]')).toHaveClassName('dollarSignH1Style')
```

Keep in mind when you shallow render the component, you will not be able to find the dollarSign element as it is currently used. You can make it shallow-render-friendly by making these changes to the component:

```diff
const PriceLockup = ({ size, price, topText, signDirection, rateText, bottomText }) => {
  // PriceLockup code

-  const DollarSign = () => {
+  const renderDollarSign = () => {
    if (size === 'large') {
      return (
        <span className={styles.dollarSignH1Style}>
          &#36;
        </span>
      )
    }
    return (
      <Text size={size === 'small' ? 'medium' : 'large'}>
        &#36;
      </Text>
    )
  }

-  const PriceValueSign = () => {
+  const renderPriceValueSign = () => {
    return (
      <Box between={size === 'large' ? 2 : 1} inline className={styles.priceValueSign}>
        {signDirection === 'left' ? <DollarSign /> : undefined}
        <span className={styles[`priceValueSign${size}`]}>{price}</span>
        {signDirection === 'right' ? <DollarSign /> : undefined}
      </Box>
    )
  }
   return (
    <Box between={3}>
      {topText && <Text size={size === 'large' ? 'large' : 'small'}>{topText}</Text>}
      <Box between={size === 'small' ? 1 : 2} inline dangerouslyAddClassName={styles.priceWrapper}>
-        <PriceValueSign />
+       {renderPriceValueSign()}
        {rateText && <Text size={size === 'large' ? 'large' : 'medium'}>{rateText}</Text>}
      </Box>
      ...
    </Box>
  )
}
```

Reply:
replaced